tracer/src/Datadog.Trace/ClrProfiler/ServerlessInstrumentation/ServerlessMiniAgent.cs

```diff
@@ -0,0 +1,164 @@
+// <copyright file="ServerlessMiniAgent.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.ClrProfiler.ServerlessInstrumentation;
+
+internal static class ServerlessMiniAgent
+{
+    private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(ServerlessMiniAgent));
+
+    internal static string GetMiniAgentPath(PlatformID os, ImmutableTracerSettings settings)
+    {
+        if (!settings.IsRunningInGCPFunctions && !settings.IsRunningInAzureFunctionsConsumptionPlan)
+        {
+            return null;
+        }
+
+        string rustBinaryPath;
+        if (Environment.GetEnvironmentVariable("DD_MINI_AGENT_PATH") != null)
```

Comment:
Two things: We have a safe wrapper for accessing env vars. There are cases when  the process doesn't have enough permissions to read them and this would throw an exception. We also have a configuration system that can read from different sources (env vars, json file, `web.config`).

This code is fine if you think you don't need either of those for GCP and AF. Just wanted to check.

Reply:
Additionally, you may want to pass the path as a method argument to make it easier to test this code, since we can't fake the env vars easily.